lib/floe/workflow/state.rb

```diff
@@ -19,27 +19,28 @@ def build!(workflow, name, payload)
         end
       end
 
-      attr_reader :workflow, :comment, :name, :type, :payload
+      attr_reader :workflow, :comment, :name, :next_state, :type, :payload
 
       def initialize(workflow, name, payload)
-        @workflow = workflow
-        @name     = name
-        @payload  = payload
-        @end      = !!payload["End"]
-        @type     = payload["Type"]
-        @comment  = payload["Comment"]
+        @workflow   = workflow
+        @name       = name
+        @payload    = payload
+        end_state   = !!payload["End"]
```

Comment:
Naming-wise this is a little weird to me, because end_state implies to me an actual state name or state object, but this is a boolean.  Perhaps

```suggestion
        is_end   = !!payload["End"]
```